quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,461 @@
+                    Object cell = currentRow.get(j);
+                    System.out.printf("Column: '%s' of type '%s', Value: '%s'%n", columns[j].getColumnName(), columns[j].getColumnType(), cell == null ? "[null]" : cell);
+                }
+                System.out.println();
+            }
+            return true;
+        } catch (DataServiceException e) {
+            System.out.printf("Server error while trying to execute query '%s' on database '%s'%n%n", query, databaseName);
+            e.printStackTrace();
+        } catch (DataClientException e) {
+            System.out.printf("Client error while trying to execute query '%s' on database '%s'%n%n", query, databaseName);
+            e.printStackTrace();
+        } catch (Exception e) {
+            System.out.printf("Unexpected error while trying to execute query '%s' on database '%s'%n%n", query, databaseName);
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private static ClientRequestProperties createClientRequestProperties(String scope) {
+        return createClientRequestProperties(scope, null);
+    }
+
+    private static ClientRequestProperties createClientRequestProperties(String scope, String timeout) {
+        ClientRequestProperties clientRequestProperties = new ClientRequestProperties();
+        clientRequestProperties.setClientRequestId(String.format("%s;%s", scope, UUID.randomUUID()));
+
+        // Tip: While not common, you can alter the request default command timeout using the below command, e.g. to set the timeout to 10 minutes, use "10m".
+        if (StringUtils.isNotBlank(timeout)) {
+            clientRequestProperties.setOption(OPTION_SERVER_TIMEOUT, timeout);
+        }
+        return clientRequestProperties;
+    }
+
+    private static void queryExistingNumberOfRows(Client kustoClient, String databaseName, String tableName) {
+        waitForUserToProceed(String.format("Get existing row count in '%s.%s'", databaseName, tableName));
+        executeQuery(kustoClient, databaseName, String.format("%s | count", tableName));
+    }
+
+    private static void alterBatchingPolicy(Client kustoClient, String databaseName, String tableName) {
+        /*
+            Tip 1: Though most users should be fine with the defaults, to speed up ingestion, such as during development and
+             in this sample app, we opt to modify the default ingestion policy to ingest data after 10 seconds have passed.
+            Tip 2: This is generally a one-time configuration.
+            Tip 3: You can also skip the batching for some files using the Flush-Immediately property, though this option should be used with care as it is inefficient.
+         */
+        waitForUserToProceed(String.format("Alter the batching policy for '%s.%s'", databaseName, tableName));
+        String command = String.format(".alter table %s policy ingestionbatching @'%s'", tableName, BATCHING_POLICY);

```

Comment:
Wholesale overwrite using .alter, or don't drop if not specified using .alter-merge?